### PR TITLE
[ci] Add release branch reporting

### PR DESCRIPTION
**Pull request by rickyyx**
Signed-off-by: rickyx <rickyx@anyscale.com>

## Why are these changes needed?
So we have test results from release branch as well. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by krfricke — APPROVED** [review]

File: release/ray_release/scripts/build_pipeline.py [hunk 1]
```diff
@@ -120,10 +120,11 @@ def main(test_collection_file: Optional[str] = None):
     if no_concurrency_limit:
         logger.warning("Concurrency is not limited for this run!")
 
-    # Report if REPORT=1 or BUILDKITE_SOURCE=schedule
+    # Report if REPORT=1 or BUILDKITE_SOURCE=schedule or it's a release branch
     report = (
         bool(int(os.environ.get("REPORT", "0")))
         or os.environ.get("BUILDKITE_SOURCE", "manual") == "schedule"
+        or branch.startswith("releases")
```
[the remark this answers is not in the document]
Comment (krfricke) [reply release/ray_release/scripts/build_pipeline.py:127]: Hm actually we may need to use `get_buildkite_repo_branch()` instead to get the wheels branch and not the test branch. Basically we should only report if both the test branch and the wheels branch are `releases/*`

**Review by krfricke — CHANGES_REQUESTED** [review]

File: release/ray_release/scripts/build_pipeline.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rickyyx) [reply release/ray_release/scripts/build_pipeline.py:127]: Sounds right to me - I wasn't sure what this test branch means tho. 

So test branch is the branch name we give to the buildkite pipeline? And the repo branch is the actual wheel branch name?

[the remark this answers is not in the document]
Comment (jjyao) [reply release/ray_release/scripts/build_pipeline.py:127]: @rickyyx how do we run release tests for release branch now? Do we provide wheel url directly?

[the remark this answers is not in the document]
Comment (rickyyx) [reply release/ray_release/scripts/build_pipeline.py:127]: > @rickyyx how do we run release tests for release branch now? Do we provide wheel url directly?

No, we are just using the latest commit (HEAD). See this pipeline https://buildkite.com/ray-project/release-tests-branch/builds/1129

**Comment by rickyyx** [comment]
Addressed comments. cc @krfricke

**Review by krfricke — APPROVED** [review]

**Pull request merged**